### PR TITLE
Reduce unexpected Integer object creation

### PR DESCRIPTION
**Pull request by miniway**
By accident, I found unexpected Integer object creation from `IntArrayList.get` investigating another performance issue. 

This might give a little gain around building join table.

Screen Shot 2020-12-02 at 2 32 23 PM

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/PagesRTreeIndex.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ private static Envelope getEnvelope(OGCGeometry ogcGeometry)
             }
         });
 
-        return matchingPositions.toIntArray(null);
+        return matchingPositions.toIntArray();
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/operator/PagesRTreeIndex.java:176]: This sounds like separate change? Can you please extract this as separate commit with proper description?

File: presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java [hunk 2]
```diff
@@ -145,7 +145,7 @@ private void appendProbeIndex(JoinProbe probe)
     {
         int position = probe.getPosition();
         verify(position >= 0);
-        int previousPosition = probeIndexBuilder.isEmpty() ? -1 : probeIndexBuilder.get(probeIndexBuilder.size() - 1);
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java:148]: Are there other collections like that with this issue?

File: presto-main/src/main/java/io/prestosql/operator/PagesRTreeIndex.java (hunk 1 above)
Comment (miniway) [reply presto-main/src/main/java/io/prestosql/operator/PagesRTreeIndex.java:176]: ok, I'll create another commit. It's just a cosmetic change as the fastutil marked `toIntArray(object)` deprecated.

File: presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java (hunk 2 above)
Comment (miniway) [reply presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java:148]: Based on profiling, I haven't found it yet. I just tried to find all `IntArrayList` usage here

**Review by sopel39 — APPROVED** [review]

**Comment by sopel39** [comment]
nice catch!

**Comment by sopel39** [comment]
merged, thanks!
